torch_xla/csrc/tensor.cpp

```diff
@@ -1410,13 +1440,10 @@ std::vector<torch::lazy::BackendDataPtr> XLATensor::FetchTensorData(
     // asynchronous operation completes.
     torch::lazy::BackendDataPtr xla_data = tensor->CurrentXlaData();
     if (xla_data == nullptr && config.force_xla_data) {
-      const torch::lazy::BackendDevice& tensor_device = tensor->GetDevice();
-      xla::Shape shape = MakeShapeWithDeviceLayout(
-          tensor->shape(), static_cast<XlaDeviceType>(tensor_device.type()));
-      xla_data =
-          WrapXlaData(xla::ComputationClient::Get()->CreateDataPlaceholder(
-              tensor_device.toString(), std::move(shape)));
-      tensor->SetXlaData(xla_data, config.sync_xla_data);
+      xla_data = tensor_data_vec[i];
```

Comment:
why not use `tensor->SetXlaData` here?

Reply:
SetXlaData also creates a new ir_value. We don't want to create ir value here because this is run after previous execution and before next execution.

Reply:
can we add a comment here?

Reply:
yeah, let me add a comment!